flink/src/test/java/org/apache/iceberg/flink/data/TestHelpers.java

```diff
@@ -0,0 +1,198 @@
+        Assert.assertTrue("Should expect a CharSequence", expected instanceof CharSequence);
+        Assert.assertEquals("string should be equal",
+            String.valueOf(expected), supplier.get().toString());
+        break;
+      case DATE:
+        Assert.assertTrue("Should expect a Date", expected instanceof LocalDate);
+        LocalDate date = ChronoUnit.DAYS.addTo(EPOCH_DAY, (int) supplier.get());
+        Assert.assertEquals("date should be equal", expected, date);
+        break;
+      case TIME:
+        Assert.assertTrue("Should expect a LocalTime", expected instanceof LocalTime);
+        int milliseconds = (int) (((LocalTime) expected).toNanoOfDay() / 1000_000);
+        Assert.assertEquals("time millis should be equal", milliseconds, supplier.get());
+        break;
+      case TIMESTAMP:
+        if (((Types.TimestampType) type).shouldAdjustToUTC()) {
+          Assert.assertTrue("Should expect a OffsetDataTime", expected instanceof OffsetDateTime);
+          OffsetDateTime ts = (OffsetDateTime) expected;
+          Assert.assertEquals("OffsetDataTime should be equal", ts.toLocalDateTime(),
+              ((TimestampData) supplier.get()).toLocalDateTime());
+        } else {
+          Assert.assertTrue("Should expect a LocalDataTime", expected instanceof LocalDateTime);
+          LocalDateTime ts = (LocalDateTime) expected;
+          Assert.assertEquals("LocalDataTime should be equal", ts,
+              ((TimestampData) supplier.get()).toLocalDateTime());
+        }
+        break;
+      case BINARY:
+        Assert.assertTrue("Should expect a ByteBuffer", expected instanceof ByteBuffer);
+        Assert.assertArrayEquals("binary should be equal", ((ByteBuffer) expected).array(), (byte[]) supplier.get());
+        break;
+      case DECIMAL:
+        Assert.assertTrue("Should expect a BigDecimal", expected instanceof BigDecimal);
+        BigDecimal bd = (BigDecimal) expected;
+        Assert.assertEquals("decimal value should be equal", bd,
+            ((DecimalData) supplier.get()).toBigDecimal());
+        break;
+      case LIST:
+        Assert.assertTrue("Should expect a Collection", expected instanceof Collection);
+        Collection<?> expectedArrayData = (Collection<?>) expected;
+        ArrayData actualArrayData = (ArrayData) supplier.get();
+        LogicalType elementType = ((ArrayType) logicalType).getElementType();
+        Assert.assertEquals("array length should be equal", expectedArrayData.size(), actualArrayData.size());
+        assertArrayValues(type.asListType().elementType(), elementType, expectedArrayData, actualArrayData);
+        break;
+      case MAP:
+        Assert.assertTrue("Should expect a Map", expected instanceof Map);
+        MapData actual = (MapData) supplier.get();
+        Assert.assertEquals("map size should be equal",
+            ((Map<?, ?>) expected).size(), actual.size());
+        Collection<?> expectedKeyArrayData = ((Map<?, ?>) expected).keySet();
```

Comment:
`keySet` isn't required to be an ordered collection, so this could easily break if keys are not returned in the same order as they are in `MapData`. This also allows maps to be equal that actually are not, like `Map('b' -> 1, 'a' -> 2)` and `Map('a' -> 1, 'b' -> 2)` if the first map's keys are reordered.

Reply:
Fixed.